scipy/stats/tests/test_multivariate.py

```diff
@@ -1429,6 +1429,427 @@ def test_samples(self):
+        mat_est_psd = _PSD(mat_est, allow_singular=False)
+        if (np.exp(mat_est_psd.log_pdet) <= 0) or (np.exp(mat_true_psd.log_pdet) <= 0):
+            return np.inf
+        trace_term = np.trace(mat_est_psd.pinv @ mat_true)
+        log_detratio = mat_est_psd.log_pdet - mat_true_psd.log_pdet
+        return (trace_term + log_detratio - len(mat_true)) / 2
+
+    @staticmethod
+    def vec(a_mat: np.ndarray) -> np.ndarray:
+        """
+        For an (m,n) array `a_mat` the output `vec(a_mat)` is an (m*n, 1)
+        array formed by stacking the columns of `a_mat` in the order in
+        which they occur in `a_mat`.
+        """
+        assert a_mat.ndim == 2
+        return a_mat.T.reshape((a_mat.size,))
+
+    def test_moments(self):
+        r"""
+        Gupta and Nagar (2000) Theorem 4.3.1 (p.135)
+        --------------------------------------------
+        The covariance of the vectorized matrix variate t-distribution equals
+        $ (V \otimes U) / (\text{df} - 2)$, where $\otimes$
+        denotes the usual Kronecker product.
+        """
+
+        df = 5
+        num_rows = 4
+        num_cols = 3
+        M = np.full((num_rows, num_cols), 0.3)
+        U = 0.5 * np.identity(num_rows) + np.full((num_rows, num_rows), 0.5)
+        V = 0.7 * np.identity(num_cols) + np.full((num_cols, num_cols), 0.3)
+        N = 10**4
+        atol = 1e-1
+
+        frozen = matrix_t(mean=M, row_spread=U, col_spread=V, df=df)
+        X = frozen.rvs(size=N, random_state=42)
+
+        relerr = self.relative_error(M, X.mean(axis=0))
+        assert_close(relerr, 0, atol=atol)
+
+        cov_vec_true = np.kron(V, U) / (df - 2)
+        cov_vec_rvs = np.cov(np.array([self.vec(x) for x in X]), rowvar=False)
+        kl = self.matrix_divergence(cov_vec_true, cov_vec_rvs)
+        assert_close(kl, 0, atol=atol)
+
+    def test_pdf_against_mathematica(self):
+        """
+        Test values generated from Mathematica 13.0.0 for Linux x86 (64-bit)
+        Release ID 13.0.0.0 (7522564, 2021120311723), Patch Level 0
+
+        mu={{1,2,3},{4,5,6}};
+        sigma={{1,0.5},{0.5,1}};
+        omega={{1,0.3,0.2},{0.3,1,0.4},{0.2,0.4,1}};
+        df=5;
+        sampleSize=10;
+        SeedRandom[42];
+        dist=MatrixTDistribution[mu,sigma,omega,df];
+        samples=RandomVariate[dist,sampleSize];
+        pdfs=PDF[dist,#]&/@samples;
+        """
+
+        df = 5
+        M = np.array([[1, 2, 3], [4, 5, 6]])
+        U = np.array([[1, 0.5], [0.5, 1]])
+        V = np.array([[1, 0.3, 0.2], [0.3, 1, 0.4], [0.2, 0.4, 1]])
+
+        atol = 1e-10
+
+        samples_m = np.array(
+            [
+                [
+                    [0.6399716994253741, 2.171718671534955, 2.5758260933527706],
+                    [4.031082477912233, 5.0216809585266375, 6.268126154787008],
+                ],
+                [
+                    [1.1648428842062322, 2.5262970999930445, 3.7813752298650685],
```

Comment:
Does Mathematica have a function to convert its arbitrary precision numbers to common double precision? These numbers have up to 17 digits but double precision typically only yields up to 15 significant digits.